src/udp.c

```diff
@@ -311,14 +392,22 @@ static int _trudpUdpIsWritable(int sd, uint32_t timeOut) {
     //}
 
     if (sendlen == -1) {
-        int err = errno;
+#if defined(TEONET_OS_WINDOWS)
+        int recv_errno = WSAGetLastError();
+#else
+        int recv_errno = errno;
+#endif
+        // TODO: Use thread safe error formatting function.
+        // TODO: On Windows use correct error formatting function.
         LTRACK_E("TrUdp",
-                 "Sending %u bytes using sendto() failed with error %d.",
-                 (uint32_t)buffer_size, err);
+                 "Sending %u bytes using sendto() failed with error %" PRId32 ": %s",
+                 (uint32_t)buffer_size, recv_errno, strerror(recv_errno));
     } else if ((size_t)sendlen != buffer_size) {
         LTRACK_E("TrUdp",
                  "Sending using sendto() sent only %u bytes of %u.",
                  (uint32_t)sendlen, (uint32_t)buffer_size);
+    } else {
+        _trudpCallUdpDataSentCallback(sendlen);
```

Comment:
I think we need define (#ifdef) here to exclude this function call in server (in teonet and teocli) (this call added for client only).